### PR TITLE
Create GitHub action to bump versions and upload ZIP files for plugin and theme

### PR DESCRIPTION
**Pull request by SantosGuillamot**
This PR aims to include a GitHub action that takes care of two things:
1. Bump version of the plugin and the `package.json` each time there is a push to `main`. Right now it is hardcoded to be always a patch.
2. Builds a ZIP file for the plugin and the theme, and adds them to the release, so they are easily downloadable. These ZIP files should only contain the files and folders necessary.

This way, users would be able to download the plugin and the theme directly from:
- Plugin: https://github.com/c4rl0sbr4v0/wp-movies-demo/releases/latest/download/wp-movies-plugin.zip
- Theme: https://github.com/c4rl0sbr4v0/wp-movies-demo/releases/latest/download/wp-movies-theme.zip

And the links will be updated with the latest changes.

I've created a [fake 0.1.1 release](https://github.com/c4rl0sbr4v0/wp-movies-demo/releases) for testing purposes, and as a way to showcase that files are downloadable. We might need to delete it before merging this Pull Request.

### CODE REVIEW

**Comment by michalczaplinski** [comment]
We should also update the README to explain that if you want to try out the demo you can just use the zips but if you want to make changes to the blocks and tinker with them then it's probably easier using Docker + `wp-env`.

**Comment by SantosGuillamot** [comment]
It seems to be an issue with how the ZIP file is created for the theme. If I copy and paste the folder into the `themes` folder, it seems to work fine. I still have to review it properly as well.

**Comment by SantosGuillamot** [comment]
There was a mismatch with some names not having a dash. It should be solved now.

**Comment by SantosGuillamot** [comment]
I've changed the base branch to `clean-repo` because this doesn't make sense until that one is merged. We don't want to create a ZIP file with the runtime code because that would be incompatible with installing the [WP Directives plugin](https://github.com/WordPress/block-hydration-experiments/).

**Comment by SantosGuillamot** [comment]
The issue with `wp-env` should be solved now.

Moreover, I updated the README with the new instructions. I tried to make it a bit more generic, so people using different local development tools can get it running, but keeping the instructions to run it with `wp-env`. Feel free to modify anything.

File: README.md [hunk 1]
```diff
@@ -37,26 +37,29 @@ It is under active development and its final public API is **very likely going t
...
+    - [Gutenberg](https://github.com/WordPress/gutenberg/releases/latest/download/gutenberg.zip)
 
-______
-##### user: `admin`
-##### password: `password`
-______
+    - [Block Interactivity Experiments](https://github.com/WordPress/block-hydration-experiments/releases/latest/download/block-interactivity-experiments.zip). This one requires Gutenberg to work.
 
-1. Install the dependencies
+    - [Movies Demo Plugin](https://github.com/c4rl0sbr4v0/wp-movies-demo/releases/latest/download/wp-movies-plugin.zip). This one requires the Block Interactiivty Experiments to work.
```
Comment (luisherranz) [inline README.md:52]:
```suggestion
    - [Movies Demo Plugin](https://github.com/c4rl0sbr4v0/wp-movies-demo/releases/latest/download/wp-movies-plugin.zip). This one requires the Block Interactivity Experiments to work.
```

File: README.md [hunk 2]
```diff
@@ -37,26 +37,29 @@ It is under active development and its final public API is **very likely going t
...
+
+    Note that, if you are using `wp-env` to run your site locally, these plugins will installed by default after running `wp-env` start. Before that, don't forget to:
```
Comment (luisherranz) [inline README.md:54]:
```suggestion
    Note that, if you are using `wp-env` to run your site locally, these plugins will be installed by default after running `wp-env` start. Before that, don't forget to:
```

**Review by luisherranz — APPROVED** [review]
I don't know if you consider this ready, but it looks great to me 🙂

**Review by michalczaplinski — APPROVED** [review]
In general everything looks great to me! 👍 

I'm using `wp-env` and I get this error on the frontend: 

Screenshot 2023-03-10 at 21 29 25

This seems related to how we're now defining the initial state on the server in https://github.com/c4rl0sbr4v0/wp-movies-demo/pull/21 and not an issue with this PR so I'm going to approve but let's wait to merge #21 first.

**Comment by SantosGuillamot** [comment]
> I'm using wp-env and I get this error on the frontend:

I'm not able to reproduce the error, even using `wp-env`. Could you confirm it is still happening with the current status of this branch?

> This seems related to how we're now defining the initial state on the server in https://github.com/c4rl0sbr4v0/wp-movies-demo/pull/21 and not an issue with this PR so I'm going to approve but let's wait to merge https://github.com/c4rl0sbr4v0/wp-movies-demo/pull/21 first.

This pull request is built on top of https://github.com/c4rl0sbr4v0/wp-movies-demo/pull/21 instead of `main`, so it shouldn't be a problem to merge this one if we consider it ready, right?

**Comment by luisherranz** [comment]
Blocks should be self-sufficient. Repeating the same state is fine if more than one block depends on it. The store is just merged on the server.

**Comment by SantosGuillamot** [comment]
Okay, that makes sense, I've added the `state` to both blocks.

**Pull request merged**